scrap.rs

```diff
@@ -0,0 +1,9 @@
+fn format_code(message: &str) {
+    println!("{}", message);
+    let path = "scrap.rs";
```

Comment:
can you find a way to put this file in a temporary directory? see https://doc.rust-lang.org/std/env/fn.temp_dir.html

Reply:
Have added as an option - however it won't allow a named temp directory and accessing temp is a bit of a pain compared to having the file appear in the project. I have included an additional function where we can write to temp - format - print to terminal - but lets chat about it on stream tomorrow. 

`fn temp_code_format(message: &str) {
    println!("{}", message);
    let dir = tempdir().unwrap();
    let mut file_path = dir.path().join("chat_code.rs");
    let path = File::create(&file_path);
    println!("{}", file_path.display());
    let _ = fs::write(&file_path, message).expect("Unable to write");
    let mut tidy = Command::new("rustfmt");
    tidy.arg(&file_path);
    tidy.status().expect("not working");
    let content = fs::read_to_string(&file_path).expect("error reading file");
    println!("{}", &content);
}`

Reply:
I understand that you want to keep the file. no need for the tmp option then, please remove.